marktplaats/query.py

```diff
@@ -87,6 +87,9 @@ def __init__(
         category=None,
         extra_attributes=None,  # EXPERIMENTAL: list of integers, just like Condition
     ):
+        if query == "" and category is None:
+            raise TypeError("Invalid arguments: When the query is empty, a category must be specified.")
```

Comment:
Might seem very nitpicky, but shouldn't this be a `ValueErrror`? https://docs.python.org/3/library/exceptions.html#TypeError:~:text=Passing%20arguments%20of%20the%20wrong%20type%20(e.g.%20passing%20a%20list%20when%20an%20int%20is%20expected)%20should%20result%20in%20a%20TypeError%2C%20but%20passing%20arguments%20with%20the%20wrong%20value%20(e.g.%20a%20number%20outside%20expected%20boundaries)%20should%20result%20in%20a%20ValueError.

Reply:
You're absolutely right, I'll change it to a `ValueError`